### ISSUE TITLE
System.ServiceModel.EndpointNotFoundException in NetTcp async streaming

### ISSUE THREAD

**Issue by KKhurin** [dotnet/wcf#915]
To repro:

```
 dnx  --configuration RETAIL run . Program2Run:Perf  Binding:NetTcp Test:Streaming Async:true
```

Within a couple of minutes we get this:

```
Exception type:   System.ServiceModel.EndpointNotFoundException
Message:          No IPEndpoints were found for host localhost.
InnerException:   <none>
StackTrace (generated):
    SP               IP               Function
    000000CA295DEA20 00007FF971073961 System_Private_ServiceModel!System.Runtime.AsyncResult.End[[System.__Canon, mscorlib]](System.IAsyncResult)+0x211
    000000CA295DEA70 00007FF9710B1586 System_Private_ServiceModel!System.ServiceModel.Channels.ServiceChannel+SendAsyncResult.End(SendAsyncResult)+0x26
    000000CA295DEAB0 00007FF9710B13B5 System_Private_ServiceModel!System.ServiceModel.Channels.ServiceChannel.EndCall(System.String, System.Object[], System.IAsyncResult)+0x1b5
    000000CA295DEB20 00007FF9710B1105 System_Private_ServiceModel!System.ServiceModel.Channels.ServiceChannelProxy+TaskCreator+<>c__DisplayClass1_0.<CreateGenericTask>b__0(System.IAsyncResult)+0x95
    000000CA295DBB60 00007FF9CA0FE75C mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)+0x3d1a9c
    000000CA295DBBA0 00007FF9C9D2CCB5 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)+0x45
    000000CA295DBBD0 00007FF9C9E9B93F mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter`1[[System.__Canon, mscorlib]].GetResult()+0x1f
    000000CA295DBC00 00007FF97106A5E8 SharedPoolsOfWCFObjects!SharedPoolsOfWCFObjects.StreamingTest`2+<TestGetStreamFromIntAsync>d__4[[System.__Canon, mscorlib],[System.__Canon, mscorlib]].MoveNext()+0x1b8
    000000CA295D9530 00007FF9CA0FE75C mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)+0x3d1a9c
    000000CA295D9570 00007FF9C9D2CCB5 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)+0x45
    000000CA295D95A0 00007FF9C9D2CC3C mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.GetResult()+0x1c
    000000CA295D95D0 00007FF9710699E1 SharedPoolsOfWCFObjects!SharedPoolsOfWCFObjects.StreamingTest`2+<RunAllScenariosAsync>d__3[[System.__Canon, mscorlib],[System.__Canon, mscorlib]].MoveNext()+0x251
    000000CA295D6F50 00007FF9CA0FE75C mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)+0x3d1a9c
    000000CA295D6F90 00007FF9C9D2CCB5 mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)+0x45
    000000CA295D6FC0 00007FF9C9D2CC3C mscorlib_ni!System.Runtime.CompilerServices.TaskAwaiter.GetResult()+0x1c
    000000CA295D6FF0 00007FF971068F32 SharedPoolsOfWCFObjects!SharedPoolsOfWCFObjects.CallStats+<CallAsyncFuncAndRecordStatsAsync>d__14.MoveNext()+0x192
```

And the corresponding server-side exception is:

```
0:041> !pe
Exception object: 00000086ba5704c0
Exception type:   System.Net.Sockets.SocketException
Message:          The I/O operation has been aborted because of either a thread exit or an application request

0:041> !clrstack
OS Thread Id: 0x398c (41)
        Child SP               IP Call Site
000000884467e888 00007ff9fb558a5c [HelperMethodFrame: 000000884467e888] 
000000884467e970 00007ff9f0191f7e *** WARNING: Unable to verify checksum for C:\windows\assembly\NativeImages_v4.0.30319_64\System.ServiceModel\df21a19f0c89a5c1ff3b57393809e148\System.ServiceModel.ni.dll
System.ServiceModel.Channels.SocketConnection.HandleReceiveAsyncCompleted() [f:\dd\ndp\cdf\src\WCF\ServiceModel\System\ServiceModel\Channels\SocketConnection.cs @ 1324]
000000884467e9b0 00007ff9efade32e System.ServiceModel.Channels.SocketConnection.OnReceiveAsync(System.Object, System.Net.Sockets.SocketAsyncEventArgs) [f:\dd\ndp\cdf\src\WCF\ServiceModel\System\ServiceModel\Channels\SocketConnection.cs @ 1297]
000000884467ea00 00007ff9f10dd630 *** WARNING: Unable to verify checksum for C:\windows\assembly\NativeImages_v4.0.30319_64\System\403e3ae044ce745ff8ee47a9891c01f7\System.ni.dll
System.Net.Sockets.SocketAsyncEventArgs.OnCompleted(System.Net.Sockets.SocketAsyncEventArgs) [f:\dd\NDP\fx\src\net\System\Net\Sockets\Socket.cs @ 8710]
000000884467ea30 00007ff9f162ce42 System.Net.Sockets.SocketAsyncEventArgs.FinishOperationAsyncFailure(System.Net.Sockets.SocketError, Int32, System.Net.Sockets.SocketFlags) [f:\dd\NDP\fx\src\net\System\Net\Sockets\Socket.cs @ 9667]
000000884467ea80 00007ff9f10dc824 System.Net.Sockets.SocketAsyncEventArgs.CompletionPortCallback(UInt32, UInt32, System.Threading.NativeOverlapped*) [f:\dd\NDP\fx\src\net\System\Net\Sockets\Socket.cs @ 9979]
000000884467ead0 00007ff9f1fbdf12 *** WARNING: Unable to verify checksum for C:\windows\assembly\NativeImages_v4.0.30319_64\mscorlib\fa8eef6f6cb67c660d71e15c5cad71b5\mscorlib.ni.dll
System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32, UInt32, System.Threading.NativeOverlapped*) [f:\dd\ndp\clr\src\BCL\system\threading\overlapped.cs @ 117]
000000884467ec70 00007ff9f3133753 [GCFrame: 000000884467ec70] 
000000884467ee68 00007ff9f3133753 [DebuggerU2MCatchHandlerFrame: 000000884467ee68] 
000000884467eff8 00007ff9f3133753 [ContextTransitionFrame: 000000884467eff8] 
000000884467f218 00007ff9f3133753 [DebuggerU2MCatchHandlerFrame: 000000884467f218] 
```

There is more flakiness happens on the server side after you let the server go in the debugger (will follow up offline on this).

**Comment by KKhurin** [comment]
We actually run out of ports... The feature is unusable on both Desktop and WCF!

**Comment by zhenlan** [comment]
Given that this issue exists in full .NET as well, we will postpone this to the future.

**Comment by KKhurin** [comment]
this is fixed with  #1752
